src/jsep.js

```diff
@@ -690,9 +698,10 @@ export class Jsep {
 	 * e.g.: `foo`, `_value`, `$x1`
 	 * Also, this function checks if that identifier is a literal:
 	 * (e.g. `true`, `false`, `null`) or `this`
+	 * @param {boolean} [allowLiterals=true] (member property must be an identifier)
 	 * @returns {jsep.Expression}
 	 */
-	gobbleIdentifier() {
+	gobbleIdentifier(allowLiterals = true) {
```

Comment:
Avoid positional boolean arguments, see 
https://ariya.io/2011/08/hall-of-api-shame-boolean-trap
https://w3ctag.github.io/design-principles/#prefer-dict-to-bool

Reply:
Also, it's a weak antipattern to have boolean arguments that default to `true`. What does this `allowLiterals` argument do exactly?

Reply:
Ah good point - I can easily flip the logic. The point is to only allow type Identifier when looking for member properties (`a.[whatever this is]`) which cannot be a 'ThisExpression', or a Literal. It would actually be easy to remove lines 726-735 from this method and then convert the returned identifier node to a literal/ThisExpr at line 524 if necessary, when it is creating the node?

Reply:
I'm not 100% sure what alternative you're proposing but if I got it right, I *think* it may be a better solution. Could we separate these changes (adding an argument to `gobbleIdentifier()` and anything that depends on it) into a separate PR and iterate on them there? That way we can quickly merge the other changes that are incontestable.

Reply:
I committed the change I was trying to describe to this PR. I also created #164 which is everything but the portion that fixes:
a.this => now matches esprima
a.true => now matches esprima